### ISSUE TITLE
return of 'Uninitialized object exists on backward branch' issue?

### ISSUE THREAD

**Issue by olagarde** [faceofcat/Tesla-Core-Lib#45]
OSX, CENT7
JRE 1.7, 1.8, 1.10
last three releases of tesla core, tesla core lib, and poweredthingies (ending 6 Aug 18)

https://pastebin.com/vxueWK1j

Probably something in my Java environment but I just can't find it.
- tried prior 3 releases in JRE 1.8 stack, same crash
- tried updating to latest 1.8 JRE, same crash
- tried 1.10 JRE compat STI for 1.8, same crash
- same with exactly one instance of JRE
- same with exactly one instance of JDK
- same with Oracle and Open JDK
- same with Oracle and Open Java runtime only
- same in fresh install on OSX and CENT7
- same in installs on base system and in fresh vbox images

For all failure cases the backtrace ends with javac bytecode validator claiming an invalid goto ref for poweredthingies pre-init bytecode generation, but this shouldn't happen if even one other user is successfully loading (which they are).  I'm at a loss.

**Comment by jriwanek** [comment]
1) `Java Version: 1.8.0_11, Oracle Corporation` That's absolutely ancient.
2) Forge doesn't work with Java 10, no point even trying that.
3) Minecraft 1.12.x was written for Java 8, no point trying it with Java 7.
No clue as to the cause, but those 3 points will atleast save you time debugging in future.

**Comment by olagarde** [comment]
True, but those are the bounds required to verify that its not things like JDK case 8046233.
FWIW Ive been running Oracle JDK 1.10 for more than a year on a double handfull of servers and packs, haven't had a problem yet that turned out to be the Java (aside from this one?).  It's actually been shockingly well behaved.;-)



-------- Original message --------From: Jasmine Iwanek <notifications@github.com> Date: 8/6/18  3:26 AM  (GMT-05:00) To: MinecraftModDevelopmentMods/Tesla-Core-Lib <Tesla-Core-Lib@noreply.github.com> Cc: olagarde <olagarde@gmail.com>, Author <author@noreply.github.com> Subject: Re: [MinecraftModDevelopmentMods/Tesla-Core-Lib] return of
  'Uninitialized object exists on backward branch' issue? (#45) 

Java Version: 1.8.0_11, Oracle Corporation That's absolutely ancient.
Forge doesn't work with Java 10, no point even trying that.
Minecraft 1.12.x was written for Java 8, no point trying it with Java 7.

No clue as to the cause, but those 3 points will atleast save you time debugging in future.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/MinecraftModDevelopmentMods/Tesla-Core-Lib","title":"MinecraftModDevelopmentMods/Tesla-Core-Lib","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jriwanek in #45: 1) `Java Version: 1.8.0_11, Oracle Corporation` That's absolutely ancient.\r\n2) Forge doesn't work with Java 10, no point even trying that.\r\n3) Minecraft 1.12.x was written for Java 8, no point trying it with Java 7.\r\nNo clue as to the cause, but those 3 points will atleast save you time debugging in future."}],"action":{"name":"View Issue","url":"https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410613530"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410613530",
"url": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410613530",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [MinecraftModDevelopmentMods/Tesla-Core-Lib] return of 'Uninitialized object exists on backward branch' issue? (#45)",
"sections": [
{
"text": "",
"activityTitle": "**Jasmine Iwanek**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@jriwanek",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"MinecraftModDevelopmentMods/Tesla-Core-Lib\",\n\"issueId\": 45,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"MinecraftModDevelopmentMods/Tesla-Core-Lib\",\n\"issueId\": 45\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410613530"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 364832175\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by jriwanek** [comment]
I'm dubious, Forge's ASM handlers *Require* Java 8 and earlier semantics, you might be able to get vanilla working on Java 10, or something not Forge, but forge itself is Java 8 or bust

**Comment by olagarde** [comment]
No problem, I can post log headers from a handfull of them tomorrow after I put jdk 1.10 back on...



-------- Original message --------From: Jasmine Iwanek <notifications@github.com> Date: 8/6/18  4:31 AM  (GMT-05:00) To: MinecraftModDevelopmentMods/Tesla-Core-Lib <Tesla-Core-Lib@noreply.github.com> Cc: olagarde <olagarde@gmail.com>, Author <author@noreply.github.com> Subject: Re: [MinecraftModDevelopmentMods/Tesla-Core-Lib] return of
  'Uninitialized object exists on backward branch' issue? (#45) 
I'm dubious, Forge's ASM handlers Require Java 8 and earlier semantics, you might be able to get vanilla working on Java 10, or something not Forge, but forge itself is Java 8 or bust

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/MinecraftModDevelopmentMods/Tesla-Core-Lib","title":"MinecraftModDevelopmentMods/Tesla-Core-Lib","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jriwanek in #45: I'm dubious, Forge's ASM handlers *Require* Java 8 and earlier semantics, you might be able to get vanilla working on Java 10, or something not Forge, but forge itself is Java 8 or bust"}],"action":{"name":"View Issue","url":"https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410630267"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410630267",
"url": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410630267",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [MinecraftModDevelopmentMods/Tesla-Core-Lib] return of 'Uninitialized object exists on backward branch' issue? (#45)",
"sections": [
{
"text": "",
"activityTitle": "**Jasmine Iwanek**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@jriwanek",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"MinecraftModDevelopmentMods/Tesla-Core-Lib\",\n\"issueId\": 45,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"MinecraftModDevelopmentMods/Tesla-Core-Lib\",\n\"issueId\": 45\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-410630267"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 364832175\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by jriwanek** [comment]
Oh, I'm not bothered one way or the other, but, go let the forge boys know they're wasting their time redoing the loader, they'll be surely pleased you've saved them the time.

**Comment by faceofcat** [comment]
according to stack overflow: `Upgrading from JDK 1.8.0_11 to 1.80_72 solved the problem for me.`. please try that.

**Comment by dshadowwolf** [comment]
How about... Minecraft Forge for Minecraft 1.12 and up requires Java 8 and uses features of Java 8 that are not available on Java 7. It is also provably incompatible with Java 9 and later.

You are also running a positively ancient version of Java 8 - for the Windows versions Mojang/Microsoft is using Java 8u51 because of issues with some older Intel chipsets and I would not be surprised if Java 8u11 was a choice on OSX for similar reasons.

To that end... Some mods actually have issues with the versions of Java that ship with vanilla Minecraft. That actually appears to be the case here, as @faceofcat has stated that upgrading to, at least, 8u72 fixes the issue.

A solid recommendation, for everyone, is that they use the latest available version of Java 8 (I believe its somewhere around 8u151 for Windows and might be the same for all platforms) and ignore later Java versions (Java 9 and up) until both Vanilla and Forge support them.

**Comment by olagarde** [comment]
True, but again, the range of versions was for testing, not regular use.  The java initially quoted worked fine (and still does) for a double handfull of servers running other various packs and several hundred mods.



-------- Original message --------From: dshadowwolf <notifications@github.com> Date: 8/7/18  8:24 AM  (GMT-05:00) To: MinecraftModDevelopmentMods/Tesla-Core-Lib <Tesla-Core-Lib@noreply.github.com> Cc: olagarde <olagarde@gmail.com>, Author <author@noreply.github.com> Subject: Re: [MinecraftModDevelopmentMods/Tesla-Core-Lib] return of
  'Uninitialized object exists on backward branch' issue? (#45) 
How about... Minecraft Forge for Minecraft 1.12 and up requires Java 8 and uses features of Java 8 that are not available on Java 7. It is also provably incompatible with Java 9 and later.
You are also running a positively ancient version of Java 8 - for the Windows versions Mojang/Microsoft is using Java 8u51 because of issues with some older Intel chipsets and I would not be surprised if Java 8u11 was a choice on OSX for similar reasons.
To that end... Some mods actually have issues with the versions of Java that ship with vanilla Minecraft. That actually appears to be the case here, as @faceofcat has stated that upgrading to, at least, 8u72 fixes the issue.
A solid recommendation, for everyone, is that they use the latest available version of Java 8 (I believe its somewhere around 8u151 for Windows and might be the same for all platforms) and ignore later Java versions (Java 9 and up) until both Vanilla and Forge support them.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/MinecraftModDevelopmentMods/Tesla-Core-Lib","title":"MinecraftModDevelopmentMods/Tesla-Core-Lib","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib"}},"updates":{"snippets":[{"icon":"PERSON","message":"@dshadowwolf in #45: How about... Minecraft Forge for Minecraft 1.12 and up requires Java 8 and uses features of Java 8 that are not available on Java 7. It is also provably incompatible with Java 9 and later.\r\n\r\nYou are also running a positively ancient version of Java 8 - for the Windows versions Mojang/Microsoft is using Java 8u51 because of issues with some older Intel chipsets and I would not be surprised if Java 8u11 was a choice on OSX for similar reasons.\r\n\r\nTo that end... Some mods actually have issues with the versions of Java that ship with vanilla Minecraft. That actually appears to be the case here, as @faceofcat has stated that upgrading to, at least, 8u72 fixes the issue.\r\n\r\nA solid recommendation, for everyone, is that they use the latest available version of Java 8 (I believe its somewhere around 8u151 for Windows and might be the same for all platforms) and ignore later Java versions (Java 9 and up) until both Vanilla and Forge support them."}],"action":{"name":"View Issue","url":"https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-411037600"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-411037600",
"url": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-411037600",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [MinecraftModDevelopmentMods/Tesla-Core-Lib] return of 'Uninitialized object exists on backward branch' issue? (#45)",
"sections": [
{
"text": "",
"activityTitle": "**dshadowwolf**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@dshadowwolf",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"MinecraftModDevelopmentMods/Tesla-Core-Lib\",\n\"issueId\": 45,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"MinecraftModDevelopmentMods/Tesla-Core-Lib\",\n\"issueId\": 45\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/MinecraftModDevelopmentMods/Tesla-Core-Lib/issues/45#issuecomment-411037600"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 364832175\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by olagarde** [comment]
Fixed, sort of. The fact that jdk versions outside 1.8 worked at all implies (requires, really) that there was a 1.8 dynlib orphaned ... somewhere. Otherwise the static forge prereq would fail, right? So I manually restored the /Library/.../JavaVM.frameworks/ and /System/.../Java/ from prior to the issue, rebooted, upgraded to 1.8.181, rebooted, and it started working again. So it probably was mixed versions, and definitely involved the base java loader dynlib (probably in /System/.../Java/ somewhere). Beyond that, who knows?
